ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -3810,11 +3810,38 @@ static void ggml_backend_sycl_graph_compute_impl(ggml_backend_sycl_context * syc
     }
 }
 
+#ifdef GGML_SYCL_GRAPH
+static bool check_node_graph_compatibility(ggml_cgraph * cgraph) {
+    for (int i = 0; i < cgraph->n_nodes; i++) {
+        ggml_tensor * node = cgraph->nodes[i];
+        switch (node->op) {
+            default:
+                break;
+            case GGML_OP_CONCAT:
+                // ggml_sycl_op_concat() does a blocking host wait after memcpy operations,
+                // but wait() can't be called on the events returned by a queue recording
+                // to a graph.
+                [[fallthrough]];
+            case GGML_OP_MUL_MAT_ID:
+                // ggml_sycl_mul_mat_id() does a blocking host wait on the sycl queue after
+                // submitting a memcpy operation, but wait() can't be called on a queue that
+                // is recording to a graph.
+#    ifndef NDEBUG
+                GGML_LOG_DEBUG("%s: disabling SYCL graphs due to unsupported node type\n", __func__);
+#    endif
```

Comment:
Nit but maybe worth having this as `GGML_LOG_INFO` and printing which node type is unsupported?

Reply:
Good idea, updated